lib/logging.py

```diff
@@ -51,6 +70,8 @@ def user_logged_out_callback(sender, request, user, **kwargs):
 # so we cannot get the IP.
 @receiver(user_login_failed)
 def user_login_failed_callback(sender, credentials, **kwargs):
-    SecurityLog.logevent(None, "login-failed","{credentials}".format(
-        credentials=credentials)
-    )
+    try:
+        SecurityLog.logevent(None, "login-failed", f"username: {credentials['username']}")
+    except KeyError:
+        # Unit tests do not have 'username' in credentials, let's not fail them for that
+        pass
```

Comment:
```suggestion
    SecurityLog.logevent(None, "login-failed", f"username: {credentials.get('username', 'none')}")
    # Unit tests do not have 'username' in credentials, let's not fail them for that
```
I would have preferred this way so that we do not miss the logs completely in case this could happen in the production environment too.